### PR TITLE
feat(mme): Add ipv6 msg updates

### PR DESCRIPTION
**Pull request by koolzz**
Signed-off-by: Nick Yurchenko <koolzz@fb.com>

## Summary

- Adds the last remaining conversions to enable IPV6 underlay layer support on agw.
- Mostly just tracking the eth1 ipv6 addr and using that whenever pdn type is ipv6

## Test Plan

- Validated on spirient

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/protos/oai/spgw_state.proto [hunk 1]
```diff
@@ -138,6 +138,8 @@ message SgwEpsBearerContext {
     repeated uint32 sdf_ids = 26;
 
     string policy_rule_name = 27;
+
+    bytes sgw_ipv6_address_s1u_s12_s4_up = 28;
```
Comment (ardzoht) [inline lte/protos/oai/spgw_state.proto:142]: we need to add state conversion and deconversion for this one, check `lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.cpp` for reference

File: lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c [hunk 2]
```diff
@@ -282,10 +282,18 @@ status_code_e spgw_handle_nw_initiated_bearer_actv_req(
   }
 
   teid_t s1_u_sgw_fteid = spgw_get_new_s1u_teid(spgw_state);
+
+  sgw_eps_bearer_ctxt_t* bearer_ctxt_p = NULL;
+  bearer_ctxt_p                        = sgw_cm_get_eps_bearer_entry(
```
Comment (pruthvihebbani) [inline lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c:287]: Add NULL check for bearer_ctxt_p

Comment (koolzz) [reply lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c:287]: If its null should I call Fatal()?

File: lte/protos/oai/spgw_state.proto (hunk 1 above)
Comment (koolzz) [reply lte/protos/oai/spgw_state.proto:142]: added

File: lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c (hunk 2 above)
Comment (pruthvihebbani) [reply lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c:287]: Throw an error message and return

Comment (koolzz) [reply lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c:287]: done

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c [hunk 3]
```diff
@@ -233,7 +233,7 @@ static uint32_t find_gtp_port_no(
   if (!spgw_config.sgw_config.ovs_config.multi_tunnel) {
     return 0;
   }
-  if ((uint32_t) enb_addr.s_addr == 0) {
+  if ((uint32_t) enb_addr.s_addr == 0 && enb_addr_ipv6 == NULL) {
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c:236]: should this be `||`?

File: lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c [hunk 4]
```diff
@@ -282,10 +282,24 @@ status_code_e spgw_handle_nw_initiated_bearer_actv_req(
...
+      &spgw_ctxt_p->sgw_eps_bearer_context_information.pdn_connection,
+      bearer_req_p->lbi);
+  if (bearer_ctxt_p == NULL) {
+    OAILOG_ERROR_UE(
+        LOG_SPGW_APP, imsi64, "Failed to retrieve bearer ctxt: \n ");
```
Comment (ardzoht) [inline lte/gateway/c/core/oai/tasks/sgw/pgw_handlers.c]:
```suggestion
        LOG_SPGW_APP, imsi64, "Failed to retrieve bearer ctxt: %u\n", bearer_req_p->lbi);
```

File: lte/gateway/python/scripts/generate_oai_config.py [hunk 5]
```diff
@@ -24,6 +24,7 @@
     get_ip_from_if,
     get_ip_from_if_cidr,
     get_ipv6_from_if,
+    get_if_ip_with_netmask,
```
Comment (ardzoht) [inline lte/gateway/python/scripts/generate_oai_config.py]: Run https://docs.magmacore.org/docs/next/lte/dev_unit_testing#format-agw for formatting

**Review by ardzoht — CHANGES_REQUESTED** [review]
minor changes to be applied

File: lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c (hunk 3 above)
Comment (koolzz) [reply lte/gateway/c/core/oai/tasks/gtpv1-u/gtp_tunnel_openflow.c:236]: nope, either one is fine. the error is only when provided ips are both null

**Review by ardzoht — APPROVED** [review]
lgtm

**Review by themarwhal — APPROVED** [review]

**Pull request merged**